### ISSUE TITLE
Select browser automation test framework

### ISSUE THREAD

**Issue by lpsinger** [nasa-gcn/gcn.nasa.gov#2133]
Select a framework for integration tests that use browser automation. These tests will have `arc sandbox` running as a sidecar process.

Some options:

- [Playwright](https://playwright.dev) (used by [Remix integration tests](https://github.com/remix-run/remix/tree/main/integration))
- [Cypress](https://www.cypress.io)
- [jest-puppeteer](https://jestjs.io/docs/puppeteer)

Acceptance tests:
- Notes on pros and cons of each of these frameworks
- A recommendation on which framework to use

**Comment by dakota002** [comment]
@lpsinger 

tldr: My vote is for playwright

**Playwright**

Easy to install:
- npm init playwright@latest
- search for playwright in vs code extensions.
- Install Playwright Test for VSCode
- Run command "Test: Install Playwright" in vscode. Will generate some example files.

Click the green Play buttons to run a test

Under Testing in VSCode, Playwrite has a menu. The `Show Browser` checkbox toggles whether the browser window is shown.

Supports multiple browsers at a time

In `playwrite.config.ts` uncomment the bottom property called `webServer`. Update the command to `npm run dev` and update the url to http://localhost:3333 (or your used port)


Test Gen is a nice feature:
Start local `npm run dev`
Once running, under testing tab, click record new. It will generate stuff based on your interactions with the site

Can run locally without much effort. Starting videos are very helpful

Basic sample: https://github.com/dakota002/gcn.nasa.gov/tree/Playwright

### Cypress

Errors out the gate.

The first demo video on their website wasn't working, which feels ironic

Con(or pro, but for me more a con): adds (effectively) a whole app into our project

Config immediately throws an error that it is invalid during its own startup process:
Stack trace: `{}` (no really, that was it. Empty brackets)

My vote is no

### Puppeteer

Pro: we are already using jest

Con: combining configs is not straight-forward-> But we got there (see preset.cjs, thanks to [this](https://github.com/argos-ci/jest-puppeteer/issues/364))

As a .js file, it throws some errors if you switch from require to import

Finally success after a bit of tweaking files and having to look into issue histories to find working code
https://github.com/dakota002/gcn.nasa.gov/tree/Puppeteer

**Comment by Courey** [comment]
I am down with Playwright. I have used both playwright and cypress. I think playwright would be great for us.

**Comment by dakota002** [comment]
Awesome, I will mark this as done and work on updating the draft PR for #2063 to include information on playwright as well as some basic coverage
